src/StudyEntities/Candidate/CandID.php

```diff
@@ -0,0 +1,20 @@
+<?php
+namespace LORIS\StudyEntities\Candidate;
+class CandID extends ValidatableIdentifier
+{
+    public function getType(): string
+    {
+        return 'CandID';
+    }
+
+    static protected function validate(string $value): bool
+    {
+        return preg_match('/[0-9]{6}/', $value) === 1;
```

Comment:
Does this validates all conditions for CandIDs?  If you have 000001, it would be validated I think, but I am not sure if it is considered a valid CandID.

Reply:
@paiva A CandID is a string of 6 digits. This does not checks for unicity or other business logic related to state.

Reply:
I think this is only intended to validate the statically analyzable / structure of the CandID. Things that require external dependencies (such as needing a database connection to check if it's valid) shouldn't be done as part of the model's validation.